Firestore/Example/Tests/SpecTests/FSTMockDatastore.mm

```diff
@@ -290,11 +293,11 @@ + (instancetype)mockDatastoreWithWorkerDispatchQueue:(FSTDispatchQueue *)workerD
   static DatabaseInfo database_info{DatabaseId{"project", "database"}, "persistence", "host",
                                     false};
 
-  FSTEmptyCredentialsProvider *credentials = [[FSTEmptyCredentialsProvider alloc] init];
+  static EmptyCredentialsProvider credentials;
```

Comment:
nit: Upon inspection, I think this is safe. But it could become not-safe if the implementation of EmptyCredsProvider changes.

Specifically, since the initWithDatabaseInfo call (below on line 298) takes a non-const pointer, it's conceivable that it could mutate EmptyCredsProvider, leading to somewhat surprising results if this method is called more than once. I don't think any of the methods of EmptyCredsProvider will actually mutate it (not even SetUserChangeListener) but if you want to be sure, either:
a) use a const pointer/ref instead of a non-const one (which may not be possible in this case since the Datastore probably does want to call methods that could potentially mutate the credentials object.)
b) allocate new ones every time like the objc version did (but then you'll need to worry about `delete`ing the pointers OR make the Datastore own the pointer.)
c) pass by value so that the FSTMockDatastore owns it.
d) Adjust the calling site (which is probably the root of the test?) to allocate the EmptyCredsProvider on the stack and then just pass in a pointer to here (thus deallocating it automatically when the test completes.) This is kinda 'dependency-injection-y'.

Since this is a test, using objects that will only be present in the test suite (EmptyCredsProvider), you could also solve this via the following sub-optimal solutions:
e) just leak memory
f) Add "It is assumed that none of the methods in this object will mutate it" to the docs of EmptyCredsProvider.

I'm not wild about e/f, but from a pragmatic point of view, they might be ok. Still, I'd recommend trying a different approach first.

I think I like (d) the best, but that does require a bit more reworking of the PR. (b) would be a close second for me.